examples/idempotency/src/working_with_idempotent_function_dataclass_output_serializer.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass
+
+from aws_lambda_powertools.utilities.idempotency import (
+    DynamoDBPersistenceLayer,
+    IdempotencyConfig,
+    idempotent_function,
+)
+from aws_lambda_powertools.utilities.idempotency.serialization.custom_dict import CustomDictSerializer
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+dynamodb = DynamoDBPersistenceLayer(table_name="IdempotencyTable")
+config = IdempotencyConfig(event_key_jmespath="order_id")  # see Choosing a payload subset section
+
+
+@dataclass
+class OrderItem:
+    sku: str
+    description: str
+
+
+@dataclass
+class Order:
+    item: OrderItem
+    order_id: int
+
+
+@dataclass
+class OrderOutput:
+    order_id: int
+
+
+order_output_serializer: CustomDictSerializer = CustomDictSerializer(
+    to_dict=lambda x: x.asdict(),
+    from_dict=lambda x: OrderOutput(**x),
```

Comment:
What do you think about creating this function as a standalone? I mean, sometimes the `lambda` syntax makes it difficult to understand the proposal and increases the cognitive load.

Maybe this:

```python
def custom_to_dict(x: Any) -> Dict:
    return dataclasses.asdict(x)

def custom_from_dict(x: Dict) -> Any:
    return OrderOutput(**x)

order_output_serializer = CustomDictSerializer(
    to_dict_func=custom_to_dict,
    from_dict_func=custom_from_dict,
) 
```

Reply:
Nice!
changed